iceoryx2-cxx/src/config.cpp

```diff
@@ -275,8 +275,8 @@ void Event::set_deadline(iox::optional<iox::units::Duration> value) && {
     value
         .and_then([&](auto value) {
             const uint64_t seconds = value.toSeconds();
-            const uint32_t nanoseconds =
-                value.toNanoseconds() - (value.toSeconds() * iox::units::Duration::NANOSECS_PER_SEC);
+            const auto nanoseconds = static_cast<uint32_t>(
+                value.toNanoseconds() - (value.toSeconds() * iox::units::Duration::NANOSECS_PER_SEC));
```

Comment:
Can you change this and replace the calculation in a similar manner to line 429 and 430 below?